specs/Workers.md

```diff
@@ -0,0 +1,1053 @@
+
+                    m_appWindow->AsyncMessageBox(scriptUrl.get(), L"Shared worker is created");
+                  }
+                  return S_OK;
+              })
+              .Get(),
+          &m_sharedWorkerCreatedToken));
+    }
+```
+
+## Retrieving Shared Workers
+
+The following example demonstrates how to fetch and display information about all the shared workers associated with the WebView2 profile.
+
+## .NET/WinRT
+```c#
+    private async void GetSharedWorkersExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+      try
+      {
+          if (SharedWorkerManager_ != null)
+          {
+              SharedWorkerManager_ = WebViewProfile.SharedWorkerManager;
+          }
+          IReadOnlyList<CoreWebView2SharedWorker> workerList = await SharedWorkerManager_.GetSharedWorkersAsync();
+          int workerCount = workerList.Count;
+          StringBuilder messageBuilder = new StringBuilder();
+          messageBuilder.AppendLine($"No of shared  workers created: {workerCount}");
+
+          for (int i = 0; i < workerCount; ++i)
+          {
+              var stringUrl = workerList[i].ScriptUrl;
+
+              messageBuilder.AppendLine($"ScriptUrl: {scriptUrl}");
+          };
+
+          MessageBox.Show(messageBuilder.ToString(), "Shared Workers", MessageBoxButton.OK);
+      }
+      catch (NotImplementedException exception)
+      {
+          MessageBox.Show(this, "GetSharedWorkersAsync Failed: " + exception.Message,
+              "Get Shared Workers Info");
+      }
+    }
+```
+## Win32 C++
+```cpp
+  void ScenarioSharedWorkerManager::GetAllSharedWorkers()
+  {
+    if (!m_sharedWorkerManager)
+    {
+        GetSharedWorkerManager();
+    }
+    CHECK_FAILURE(m_sharedWorkerManager->GetSharedWorkers(
+        Callback<ICoreWebView2GetSharedWorkersCompletedHandler>(
+            [this](
+                HRESULT error,
+                ICoreWebView2SharedWorkerCollectionView* workersCollection) -> HRESULT
+            {
+               if(workersCollection)
+               {
+                  UINT32 workersCount = 0;
+                  CHECK_FAILURE(workersCollection->get_Count(&workersCount));
+
+                  std::wstringstream message{};
+                  message << L"No of shared workers created: " << workersCount << std::endl;
+
+                  for (UINT32 i = 0; i < workersCount; i++)
+                  {
+                      Microsoft::WRL::ComPtr<ICoreWebView2SharedWorker> sharedWorker;
+                      CHECK_FAILURE(workersCollection->GetValueAtIndex(i, &sharedWorker));
+
+                      wil::com_ptr<ICoreWebView2Worker> worker;
+                      if (SUCCEEDED(sharedWorker->QueryInterface(IID_PPV_ARGS(&worker))))
+                      {
+                          wil::unique_cotaskmem_string scriptUrl;
+                          CHECK_FAILURE(worker->get_ScriptUrl(&scriptUrl));
+
+                          message << L"ScriptUrl: " << scriptUrl.get();
```

Comment:
We want sample code that demonstrates doing something we expect end developers to actually write code for in their apps. The code you have here is more of sandbox / playground sort of thing where someone can try out the API rather than what we would expect the API to actually be used for.

Reply:
As discussed, I added sample code for apps to utilize this worker monitoring for posting messages to worker for one sample of each worker. Please do suggest if it is not conveying it properly.